st2client/st2client/commands/auth.py

```diff
@@ -70,6 +73,76 @@ def run_and_print(self, args, **kwargs):
                               attributes=self.display_attributes, json=args.json, yaml=args.yaml)
 
 
+class LoginCommand(resource.ResourceCommand):
+    display_attributes = ['user', 'token', 'expiry']
+
+    def __init__(self, resource, *args, **kwargs):
+
+        kwargs['has_token_opt'] = False
+
+        super(LoginCommand, self).__init__(
+            resource, kwargs.pop('name', 'create'),
+            'Authenticate user, acquire access token, and update CLI config directory',
+            *args, **kwargs)
+
+        self.parser.add_argument('username',
+                                 help='Name of the user to authenticate.')
+
+        self.parser.add_argument('-p', '--password', dest='password',
+                                 help='Password for the user. If password is not provided, '
+                                      'it will be prompted.')
+        self.parser.add_argument('-l', '--ttl', type=int, dest='ttl', default=None,
+                                 help='The life span of the token in seconds. '
+                                      'Max TTL configured by the admin supersedes this.')
+        self.parser.add_argument('-w', '--write-real-password', action='store_true', default=False,
+                                 dest='write_password',
+                                 help='Write the real (plain-text) password to the config file')
+
+    def run(self, args, **kwargs):
+
+        if not args.password:
+            args.password = getpass.getpass()
+        instance = self.resource(ttl=args.ttl) if args.ttl else self.resource()
+
+        manager = self.manager.create(instance, auth=(args.username, args.password), **kwargs)
+
+        cli = BaseCLIApp()
+
+        logging.basicConfig(
+            level=logging.CRITICAL,
+            handlers=[logging.NullHandler]
+        )
+        cli.LOG = logging.getLogger("st2client.base")
+        cli._cache_auth_token(token_obj=manager)
+
+        # Update existing configuration with new credentials
+        config_file = expanduser('~/.st2/config')
```

Comment:
This is a good idea, and I used the function `_get_config_file_path` in BaseCLI to ensure this consistency. Implemented in 683cb4d

This is how it works now (note if the `--config` option is omitted, that still works and writes to the default location:

```
(virtualenv)vagrant@st2dev:~/st2/st2client$ st2 --config /tmp/st2config login st2admin --password Password1!
Logged in as st2admin
(virtualenv)vagrant@st2dev:~/st2/st2client$ cat /tmp/st2config
[credentials]
username = st2admin
password = notarealpassword
```